Info-Orbs/src/core/wifiWidget.cpp

```diff
@@ -15,36 +20,58 @@ void WifiWidget::setup() {
   m_manager.selectScreen(0);
   display.drawCentreString("Connecting" + m_connectionString, 120, 80, 1);
 
+  WiFi.mode(WIFI_STA); // For WiFiManager explicitly set mode to station, esp defaults to STA+AP
 
-  m_manager.selectScreen(1);
-  display.drawCentreString("Connecting to", 120, 80, 1);
-  display.drawCentreString("WiFi..", 120, 100, 1);
-  display.drawCentreString(WIFI_SSID, 120, 130, 1);
-
-  // Serial.println("Connecting to WiFi..");
-
-  WiFi.begin(WIFI_SSID, WIFI_PASS);
-
-  Serial.println("Connecting to WiFi..");
+  // reset settings - wipe stored credentials for testing
+  // these are stored by the esp WiFi library
+  // wifimgr.resetSettings();
 
+  // WiFiManager automatically connects using saved credentials,
+  // if connection fails, it starts an access point with the specified name ( "AutoConnectAP"),
+  // if empty will auto generate SSID, if password is blank it will be anonymous AP (wm.autoConnect())
+  // then goes into a blocking loop awaiting configuration and will return success result
 
+  // Set WiFiManager to non-blocking so status and info can be displayed
+  wifimgr.setConfigPortalBlocking(false);
+  // wm.setConfigPortalTimeout(60);
 
+  // wm.autoConnect(); // auto generated AP name from chipid
+  // wm.autoConnect("AutoConnectAP"); // anonymous AP
+  // wifimgr.autoConnect("AutoConnectAP","password"); // password protected AP
+  if(wifimgr.autoConnect("Info-Orbs")){
+    Serial.print("WifiManager connected.");
+  }
+  else {
+    m_configPortalRunning = true;
+    Serial.println("Configuration portal running.");
+    m_manager.selectScreen(1);
+    display.drawCentreString("Configure WiFi", 120, 80, 1);
+    display.drawCentreString("by browsing to:", 120, 100, 1);
+    display.drawCentreString("192.168.4.1", 120, 120, 1);
```

Comment:
Should this be hardcoded, or coming from the wifimanager?

Reply:
WiFiManager defaults to 192.168.4.1. This default can be overridden, so it would be better pull the IP address frim WiFiManager. It seemed a fairly safe shortcut to hardcode it, but yes it's still a bad practice shortcut.